### ISSUE TITLE
Log software + packages versions

### ISSUE THREAD

**Issue by jc-cisneros** [gentzkow/template#71]
In this issue we will test the `write_version_log` feature from gslab_make (see https://github.com/gslab-econ/gslab_make/issues/56).

Task list

- [x] Add the new feature to the make scripts in each submodule.
- [x] Check that a `versions.log` file is created in each submodule's `log` folder.
- [x] Check that `versions.log` correctly prints the versions from the active environment.  
- [x] Robustness tests (i.e., the script should produce a result for all possible user behavior and throw appropriate error messages when relevant)

**Comment by jc-cisneros** [comment]
@gentzkow, following the latest conversations on package versions (see https://github.com/gslab-econ/gslab_make/issues/56), I added a `write_versions_log` feature to produce a log file that lists all the software + packages used in a conda environment, their versions and a timestamp. This feature can be added to any project using `gslab_make` so that we can always check the versions used in any commit. A caveat of the current code is that it requires a conda environment to be active (i.e, if a user has all the dependencies installed locally, the `versions.log` file would not be created). 

The `make.log` files now include a new line when the version logs are correctly created:
<img width="598" alt="Screen Shot 2022-11-09 at 12 01 02 AM" src="https://user-images.githubusercontent.com/108437517/200775681-7d8cf6c0-7328-4878-a823-3e53b0232a95.png">

An example of how a `versions.log` file for a module looks now can be found [on the latest commit](https://github.com/gentzkow/template/blob/c6e6f3a258590cca7782db0088a403304a3fc87d/analysis/log/versions.log).

If this sounds good, I can proceed with the double PR (on both `template` and `gslab_make`).

**Comment by gentzkow** [comment]
Great! This looks very good.

I'm thinking we may want to incorporate this into `template` and then make the default to *not* specify explicit version numbers in the setup file. We can note in a comment in that file that users can add explicit version numbers if they want to freeze a package at a particular revision or if they're getting conflicts caused by package updates.

**Comment by jc-cisneros** [comment]
On [a98903e](https://github.com/gentzkow/template/commit/a98903ecafc74ce12ebafa088864e5aff6603ac9), I updated the `conda_env.yaml` to reflect the new default behavior per https://github.com/gentzkow/template/issues/71#issuecomment-1309251979. I will proceed with the PR for the issue.

**Comment by jc-cisneros** [comment]
Summary:

On this issue we tested the a new gslab_make feature to log the package versions. The feature worked as expected and was added to the [gslab_make master branch](https://github.com/gslab-econ/gslab_make/pull/57).

Flagging some of the lessons and opportunities for future updates:

- The `write_versions_log.py` feature currently returns the output of running `conda list` with the active environment and a timestamp. We could add some extra formatting to differentiate Python, R, and Julia packages from each other (see https://github.com/gentzkow/template/pull/72#discussion_r1020986095).
- The current behavior of this function is to create an identical `versions.log` file for each module. In the future we can try to produce a `versions.log` that prints only the packages that were used in each module or script  (see https://github.com/gentzkow/template/pull/72#discussion_r1020986973).
